pykusto/_src/functions.py

```diff
@@ -1120,6 +1120,13 @@ def week_of_year() -> _NumberExpression:
         """
         raise NotImplementedError()
 
+    @staticmethod
+    def day_of_week(a_date: DatetimeType) -> '_TimespanExpression':
+        """
+        https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/dayofweekfunction
+        """
+        return a_date.day_of_week()
```

Comment:
`a_date` could be a Python datetime object, and then this will not work.
OMG I just realized this bug exists in many other places in this file...
OK, I opened #178 for this, but at least let's not add to the problem.

Reply:
Also add a test for this